src/Nethermind/Nethermind.Consensus.AuRa/Withdrawals/WithdrawalProcessor.cs

```diff
@@ -0,0 +1,55 @@
+// SPDX-FileCopyrightText: 2023 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Numerics;
+using Nethermind.Consensus.AuRa.Contracts;
+using Nethermind.Consensus.Withdrawals;
+using Nethermind.Core;
+using Nethermind.Core.Specs;
+using Nethermind.Logging;
+
+namespace Nethermind.Consensus.AuRa.Withdrawals;
+
+public class WithdrawalProcessor : IWithdrawalProcessor
+{
+    private readonly IWithdrawalContract _contract;
+    private readonly ILogger _logger;
+
+    public WithdrawalProcessor(IWithdrawalContract contract, ILogManager logManager)
+    {
+        ArgumentNullException.ThrowIfNull(logManager);
+
+        _contract = contract ?? throw new ArgumentNullException(nameof(contract));
+        _logger = logManager.GetClassLogger();
+    }
+
+    public void ProcessWithdrawals(Block block, IReleaseSpec spec)
+    {
+        if (!spec.WithdrawalsEnabled)
+            return;
+
+        if (_logger.IsTrace) _logger.Trace($"Applying withdrawals for block {block}");
+
+        if (block.Withdrawals is not null) // This check looks redundant
+        {
+            var amounts = Array.Empty<ulong>();
+            var addresses = Array.Empty<Address>();
```

Comment:
Either:
1. Initialize new Arrays
2. Initialize stack Spans (would need to update ABI code)
3. Rent arrays from array pools (simplest?)